### PR TITLE
Only for review

### PR DESCRIPTION
**Pull request by byb121**

### CODE REVIEW

File: setup.sh [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+export PATH=`echo $INST_PATH/bin:$BB_INST/bin:$PATH | perl -pe 's/:\$//;'`
+export MANPATH=`echo $INST_PATH/man:$BB_INST/man:$INST_PATH/share/man:$MANPATH | perl -pe 's/:\$//;'`
+export PERL5LIB=`echo $INST_PATH/lib/perl5:$PERL5LIB | perl -pe 's/:\$//;'`
+set -u
+
+## INSTALL CPANMINUS
+set -eux
+curl -sSL https://cpanmin.us/ > $SETUP_DIR/cpanm
+perl $SETUP_DIR/cpanm --no-wget --no-interactive --notest --mirror http://cpan.metacpan.org -l $INST_PATH App::cpanminus
+rm -f $SETUP_DIR/cpanm
```
Comment (keiranmraine) [inline setup.sh]: remove this, cpanm should be installed via package in docker (if not already present)

File: setup.sh [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+CPANM=`which cpanm`
+
+# Install sPlot
+$CPANM --no-wget --no-interactive --notest --mirror http://cpan.metacpan.org --notest -l $INST_PATH --installdeps $SCRIPT_PATH/perl/
+$CPANM --no-wget --no-interactive --notest --mirror http://cpan.metacpan.org --notest -l $INST_PATH $SCRIPT_PATH/perl/
```
Comment (keiranmraine) [inline setup.sh]: Tests should run for the actual package code

File: setup.sh [hunk 3]
```diff
@@ -0,0 +1,85 @@
+#!/bin/bash
+
+##########LICENCE##########
+# Copyright (c) 2014-2019 Genome Research Ltd.
```
Comment (keiranmraine) [inline setup.sh]: `2019` only, not on GitHub prior to this

Comment (keiranmraine) [reply setup.sh]: All instances

File: perl/Makefile.PL [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+WriteMakefile(
+  NAME          => 'sPlot',
+  LICENSE       => 'agpl_3', # http://search.cpan.org/~dagolden/CPAN-Meta-2.142690/lib/CPAN/Meta/Spec.pm#license
+  VERSION_FROM  => 'lib/Sanger/CGP/splot.pm',
+  EXE_FILES     => [qw( bin/convert_vcfToJson.pl )],
+
+  PREREQ_PM     => {
+                    'Const::Fast' => '0.014',
+                    'JSON' =>  2.90,
```
Comment (keiranmraine) [inline perl/Makefile.PL]: Is this all that was needed in the end?

File: perl/t/2_pl_compile.t [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+# details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+##########LICENCE##########
+
+# this is a catch all to ensure all modules do compile
+# added as lots of 'use' functionality is dynamic in pipeline
+# and need to be sure that all modules compile.
+# simple 'perl -c' is unlikely to work on head scripts any more.
```
Comment (keiranmraine) [inline perl/t/2_pl_compile.t]: Delete these lines. Code has been c/p from other projects

File: setup.sh [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+
+CPU=`grep -c ^processor /proc/cpuinfo`
+if [ $? -eq 0 ]; then
+  if [ "$CPU" -gt "6" ]; then
+    CPU=6
+  fi
+else
+  CPU=1
+fi
+echo "Max compilation CPUs set to $CPU"
```
Comment (keiranmraine) [inline setup.sh]: Not used, remove

**Review by keiranmraine — CHANGES_REQUESTED** [review]

File: perl/Makefile.PL (hunk 4 above)
Comment (byb121) [reply perl/Makefile.PL]: Needed a few more, but didn't know this until tested the perl module.

File: perl/Makefile.PL [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+                    # below are only required for testing the module
+                    'Try::Tiny' => 0.30,
+                    'IPC::System::Simple' => 1.25,
```
Comment (keiranmraine) [inline perl/Makefile.PL]: We can go ever cleaner in the installed version:

https://metacpan.org/pod/ExtUtils::MakeMaker#TEST_REQUIRES

**Review by keiranmraine — CHANGES_REQUESTED** [review]
We should do it properly really, need to review old things on subsequent releases to fix up this type of thing.

File: perl/Makefile.PL (hunk 7 above)
Comment (byb121) [reply perl/Makefile.PL]: Ha, I knew there must be such thing in Perl, but after googling, didn't find the answer in the end, hence the comment....

**Pull request closed without merging**